packages/angular_devkit/core/src/jobs/create-job.ts

```diff
@@ -0,0 +1,230 @@
+        logger,
+        progress,
+        createChannel(name: string) {
+          if (channels.has(name)) {
+            throw new ChannelAlreadyExistException(name);
+          }
+          const channelSubject = new Subject<JsonValue>();
+          channelSubject.subscribe(
+            message => {
+              subject.next({ kind: JobEventKind.ChannelMessage, description, name, message });
+            },
+            error => {
+              subject.next({ kind: JobEventKind.ChannelError, description, name, error });
+              // This can be reopened.
+              channels.delete(name);
+            },
+            () => {
+              subject.next({ kind: JobEventKind.ChannelComplete, description, name });
+              // This can be reopened.
+              channels.delete(name);
+            },
+          );
+
+          channels.set(name, channelSubject);
+
+          return channelSubject;
+        },
+      };
+
+      const result = fn(argument, newContext);
+      // If the result is a promise, simply wait for it to complete before reporting full progress
+      // and the result.
+      if (isPromise(result)) {
+        result.then(result => {
+          subject.next({ kind: JobEventKind.Output, description, output: result });
+          progress(1);
+          subject.next({ kind: JobEventKind.End, description });
+          subject.complete();
+        }, err => subject.error(err));
+      } else if (isObservable(result)) {
+        subscription = (result as Observable<O>).subscribe(
+          (output: O) => subject.next({ kind: JobEventKind.Output, description, output }),
+          error => subject.error(error),
+          () => {
+            progress(1);
+            subject.next({ kind: JobEventKind.End, description });
+            subject.complete();
+          },
+        );
+
+        return subscription;
+      } else {
+        // If it's a scalar value, report it synchronously.
+        subject.next({ kind: JobEventKind.Output, description, output: result as O });
+        progress(1);
+        subject.next({ kind: JobEventKind.End, description });
+        subject.complete();
+      }
+    });
+  };
+
+  return Object.assign(handler, { jobDescription: options });
+}
+
+
+/**
+ * Lazily create a job using a function.
+ * @param loader A factory function that returns a promise/observable of a JobHandler.
+ * @param options Same options as createJob.
+ */
+export function createJobFactory<A extends JsonValue, I extends JsonValue, O extends JsonValue>(
+  loader: () => Promise<JobHandler<A, I, O>>,
+  options: Partial<JobDescription>,
+): JobHandler<A, I, O> {
+  const handler = (argument: A, context: JobHandlerContext<A, I, O>) => {
+    return from(loader())
+      .pipe(switchMap(fn => fn(argument, context)));
+  };
+
+  return Object.assign(handler, { jobDescription: options });
+}
+
+
+/**
+ * Creates a job that logs out input/output events to another Job. The events are still
```

Comment:
Reading the code it looks like it's logging out events **from** another job, not **to** another job.

Reply:
Replaced to `of`.